packages/dds/tree/src/test/shared-tree-core/branch.spec.ts

```diff
@@ -0,0 +1,390 @@
+		assert.equal(changeEventCount, 1);
+	});
+
+	it("emit change events during but not after committing a transaction", () => {
+		// Create a branch and count the change events emitted
+		let changeEventCount = 0;
+		const branch = create((c) => {
+			changeEventCount += 1;
+		});
+		// Begin a transaction
+		branch.startTransaction();
+		// Ensure that the correct change is emitted when applying changes in a transaction
+		change(branch);
+		assert.equal(changeEventCount, 1);
+		change(branch);
+		assert.equal(changeEventCount, 2);
+		// Commit the transaction. No change event should be emitted since the commits, though squashed, are still equivalent
+		branch.commitTransaction();
+		assert.equal(changeEventCount, 2);
+	});
+
+	it("emit a change event after aborting a transaction", () => {
+		// Create a branch and count the change events emitted
+		let changeEventCount = 0;
+		const branch = create(() => (changeEventCount += 1));
+		// Begin a transaction
+		branch.startTransaction();
+		// Apply a couple of changes to the branch
+		change(branch);
+		change(branch);
+		// Ensure the the correct number of change events have been emitted so far
+		assert.equal(changeEventCount, 2);
+		// Abort the transaction. A new change event should be emitted since the state rolls back to before the transaction
+		branch.abortTransaction();
+		assert.equal(changeEventCount, 3);
+	});
+
+	it("do not emit a change event after aborting an empty transaction", () => {
+		// Create a branch and count the change events emitted
+		let changeEventCount = 0;
+		const branch = create(() => (changeEventCount += 1));
+		// Start and immediately abort a transaction
+		branch.startTransaction();
+		branch.abortTransaction();
+		assert.equal(changeEventCount, 0);
+	});
+
+	it("correctly report whether they are in the middle of a transaction", () => {
+		// Create a branch and test `isTransacting()` during two transactions, one nested within the other
+		const branch = create();
+		assert.equal(branch.isTransacting(), false);
+		branch.startTransaction();
+		assert.equal(branch.isTransacting(), true);
+		branch.startTransaction();
+		assert.equal(branch.isTransacting(), true);
+		branch.abortTransaction();
+		assert.equal(branch.isTransacting(), true);
+		branch.commitTransaction();
+		assert.equal(branch.isTransacting(), false);
+	});
+
+	it("squash their commits when committing a transaction", () => {
+		// Create a new branch and start a transaction
+		const branch = create();
+		branch.startTransaction();
+		// Apply two changes to it
+		const tag1 = change(branch);
+		const tag2 = change(branch);
+		// Ensure that the commits are in the correct order with the correct tags
+		assertHistory(branch, tag1, tag2);
+		// Commit the transaction and ensure that there is now only one commit on the branch
+		branch.commitTransaction();
+		assert.equal(branch.getHead().parent?.revision, nullRevisionTag);
+	});
+
+	it("rollback their commits when aborting a transaction", () => {
```

Comment:
nit: These editing/transaction related tests aren't specific to branches so this feels like the wrong place for them. I think the itView tests you have are probably sufficient.

Reply:
What do you mean? Transactions are an API on `SharedTreeBranch`, and this file is testing the capabilities of `SharedTreeBranch`.

Reply:
Oh, you're saying that we already have coverage because of the transaction tests a layer up in `SharedTree.spec.ts`. Well, we could say that about `rebaseOnto` and `merge` too :P (which is why we didn't have this test suite in the first place when branches were introduced). But I think it's valuable to unit test at the lowest layer, since we easily can. I'm okay with double test coverage.